engine/src/main/java/org/entando/entando/web/user/UserController.java

```diff
@@ -150,12 +140,25 @@ public void setUserValidator(UserValidator userValidator) {
     @RequestMapping(value = "/{username:.+}", method = RequestMethod.DELETE, produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<?> deleteUser(@PathVariable String username) throws ApsSystemException {
         logger.debug("deleting {}", username);
+
+        if (!userValidator.isValidDeleteUser(username)) {
+            throw new ValidationGenericException(createDeleteAdminError(username));
+        }
+
         this.getUserService().removeUser(username);
         Map<String, String> result = new HashMap<>();
         result.put("code", username);
         return new ResponseEntity<>(new RestResponse(result), HttpStatus.OK);
     }
 
+    private BindingResult createDeleteAdminError(@PathVariable String username) {
```

Comment:
This needs test coverage for master

Reply:
Added missing unit tests and made some small improvements to the code